app/assets/javascripts/discourse/app/controllers/share-topic.js

```diff
@@ -1,23 +1,42 @@
 import Controller from "@ember/controller";
 import { action } from "@ember/object";
 import { getAbsoluteURL } from "discourse-common/lib/get-url";
-import discourseComputed from "discourse-common/utils/decorators";
+import discourseComputed, { observes } from "discourse-common/utils/decorators";
 import { ajax } from "discourse/lib/ajax";
 import { extractError } from "discourse/lib/ajax-error";
 import Sharing from "discourse/lib/sharing";
 import showModal from "discourse/lib/show-modal";
 import { bufferedProperty } from "discourse/mixins/buffered-content";
 import ModalFunctionality from "discourse/mixins/modal-functionality";
 import I18n from "I18n";
+import Category from "discourse/models/category";
 
 export default Controller.extend(
   ModalFunctionality,
   bufferedProperty("invite"),
   {
+    topic: null,
+    restrictedGroups: null,
+
     onShow() {
       this.set("showNotifyUsers", false);
     },
 
+    @observes("topic")
```

Comment:
We are trying to migrate away from `observes`. Is there any way this code can be called when an event is triggered, say the share interface shows up?

Reply:
The issue was that the topic wasn't present when onShow was first fired. See my subsequent commit for a possible solution to this issue. Is there a better/cleaner way to do this?